047_h_L552/047_h_L552_unsafe.py

```
@@ -0,0 +1,1176 @@
+
+
+def edit_entry(key):
+    db = get_db_by_key(key)
+    dbfile = "{root}/data/{db}.json".format(root=cfg.ROOT, db=db)
+    d = OrderedDict()
+    d[key] = hdict[key]
+    tmpfile = "{root}/tmp/temp.{pid}.json".format(root=cfg.ROOT, pid=os.getpid())
+    with open(tmpfile, "w") as f:
+        json.dump(d, f, indent=4)
+    assert os.path.isfile(tmpfile)
+    os.system("{ed} {fname}".format(ed=cfg.EDITOR, fname=tmpfile))
```

Ruff (0.14.1)

Starting a process with a shell, possible injection detected

(S605)